src/openforms/js/components/admin/form_design/FormLogic.js

```diff
@@ -385,6 +341,137 @@ const Rule = ({
 
         <ActionSet name="actions" actions={actions} onChange={onChange} errors={errors.actions} />
       </div>
+    </>
+  );
+};
+
+RuleBody.propTypes = {
+  isCreate: PropTypes.bool.isRequired,
+  isAdvanced: PropTypes.bool.isRequired,
+  jsonLogicTrigger: PropTypes.object,
+  actions: PropTypes.arrayOf(PropTypes.object),
+  displayAdvancedOptions: PropTypes.bool,
+  description: PropTypes.string,
+  triggerFromStepIdentifier: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  errors: PropTypes.object,
+  mayGenerateDescription: PropTypes.bool.isRequired,
+};
+
+const Rule = ({
+  isCreate,
+  _logicType,
+  description,
+  _mayGenerateDescription,
+  order,
+  jsonLogicTrigger,
+  triggerFromStep: triggerFromStepIdentifier,
+  actions,
+  isAdvanced,
+  onChange,
+  onDelete,
+  errors = {},
+}) => {
+  const intl = useIntl();
+  const [displayAdvancedOptions, setDisplayAdvancedOptions] = useState(false);
+
+  const deleteConfirmMessage = intl.formatMessage({
+    description: 'Logic rule deletion confirm message',
+    defaultMessage: 'Are you sure you want to delete this rule?',
+  });
+
+  // if no logicType has been set yet, we first present the type selection before the
+  // actual rule can be set up.
+  if (!_logicType) {
+    return (
+      <LogicTypeSelection
+        onChange={selectedType => onChange({target: {name: '_logicType', value: selectedType}})}
+        onCancel={onDelete}
+      />
+    );
+  }
+
+  const boundaryErrorMessage = (
+    <div className="logic-rule-error">
+      <div className="logic-rule-error__text">
+        <FormattedMessage
+          description="Unexpected error message in logic rule"
+          defaultMessage="Something went wrong while rendering this logic rule! For debugging, here is the JSON of the logic rule:"
+        />
+      </div>
+
+      <div className="logic-rule-error__json">
+        {JSON.stringify({
+          jsonLogicTrigger,
+          triggerFromStepIdentifier,
+          order,
+          actions,
+          isAdvanced,
+        })}
```

Comment:
wrapping this in `<code><pre>...</pre></code>` might make it look a tad better?

Reply:
![Screenshot from 2023-09-08 12-15-04](https://github.com/open-formulieren/open-forms/assets/19154114/e2a486a0-c533-4faf-ba71-e35cd71f886d)
:tada: 
(I left out the `<pre>` because it was making it go off the page)